### PR TITLE
castor, gitAndTools.git-interactive-rebase-tool, the-way: disable checks

### PR DESCRIPTION
**Pull request by zowoq**
###### Motivation for this change

These three have been broken since https://github.com/NixOS/nixpkgs/pull/95622 as we can't pass `test-threads` twice.

Just disabling the checks for now as 20.09 branch off is close and a proper fix will probably cause a rust rebuild.

```
Running cargo test --release --target x86_64-unknown-linux-gnu --frozen --  --test-threads=1
   Compiling git-interactive-rebase-tool v1.2.1 (/build/source)
    Finished release [optimized] target(s) in 6.78s
     Running target/x86_64-unknown-linux-gnu/release/deps/interactive_rebase_tool-2ed5e41c7143da47
error: Option 'test-threads' given more than once
error: test failed, to rerun pass '--bin interactive-rebase-tool'
```

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by fgaz — APPROVED** [review]
Looks fine. How do we remember to revert this when the issue is fixed?

**Comment by masaeedu** [comment]
@zowoq If I remember correctly, the tests for `git-interactive-rebase-tool` sporadically fail unless `--test-threads` is set to `1`. See https://github.com/MitMaro/git-interactive-rebase-tool/issues/172#issuecomment-581161373. I don't know if this has since been fixed, but if it hasn't, do the current settings mean that a user with `NIX_BUILD_CORES` set to e.g. 4 will experience failing tests while trying to build this package?

File: pkgs/applications/networking/browsers/castor/default.nix [hunk 1]
```diff
@@ -48,4 +49,3 @@ rustPlatform.buildRustPackage rec {
     maintainers = with maintainers; [ fgaz ];
   };
 }
-
```
Comment (fgaz) [inline pkgs/applications/networking/browsers/castor/default.nix:51]: Random whitespace change

**Review by masaeedu — APPROVED** [review]

**Comment by wkral** [comment]
I noticed that `jwt-cli` is also failing because of the `--test-threads` argument: [Hydra log](https://hydra.nixos.org/build/126783964/nixlog/1).

I can disable checks and backport to 20.09 for ZHF, but I'm a little unclear are you not expecting the proper fix to land for 20.09?

**Comment by zowoq** [comment]
I've opened a draft with a fix, I'll test/finish it later today. https://github.com/NixOS/nixpkgs/pull/97603

___

> I can disable checks

Please do. Feel free to ping me, I'll merge and cherrypick it to 20.09.

> are you not expecting the proper fix to land for 20.09?

We can backport the fix after it lands in master.

**Comment by wkral** [comment]
Looking at your fix and at the `jwt-cli` failure it's actually a little different. It seams that maybe the `jwt-cli` test is trying to parse the --test-threads argument which should be consumed by cargo. I'm not sure why exactly, but I don't think the single threaded test fix you're proposing will fix this particular issue.

I'll look into why it's happening, but if you've seen this before then I'd appreciate the help.

**Comment by wkral** [comment]
Figured it out, propsoed an upstream fix and created #97615. Thanks for the help @zowoq.

**Pull request merged**